python/lsst/ip/diffim/dcrModel.py

```diff
@@ -545,9 +486,52 @@ def calculateNoiseCutoff(self, maskedImage, statsCtrl, sigma, bufferSize,
         convergeMask = mask.getPlaneBitMask(convergenceMaskPlanes)
 
         backgroundPixels = mask[bboxShrink].array & (statsCtrl.getAndMask() | convergeMask) == 0
-        noiseCutoff = sigma*np.std(maskedImage[bboxShrink].image.array[backgroundPixels])
+        noiseCutoff = np.std(maskedImage[bboxShrink].image.array[backgroundPixels])
         return noiseCutoff
 
+    def applyImageThresholds(self, maskedImage, highThreshold=None, lowThreshold=None,
+                             regularizationWidth=2):
+        """Restrict image values to be between upper and lower limits.
+
+        This method flags all pixels in an image that are outside of the given
+        threshold values. The threshold values are taken from a reference image,
+        so noisy pixels are likely to get flagged. In order to exclude those
+        noisy pixels, the array of flags is eroded and dilated, which removes
+        isolated pixels outside of the thresholds from the list of pixels to be
+        modified. Pixels that remain flagged after this operation have their
+        values set to the appropriate upper or lower threshold value.
+
+        Parameters
+        ----------
+        maskedImage : `lsst.afw.image.MaskedImage`
+            The image to apply the thresholds to.
+            The image plane values will be modified in place.
+        highThreshold : `numpy.ndarray`, optional
+            Array of upper limit values for each pixel of ``maskedImage``.
+        lowThreshold : `numpy.ndarray`, optional
+            Array of lower limit values for each pixel of ``maskedImage``.
+        regularizationWidth : `int`, optional
+            Minimum radius of a region to include in regularization, in pixels.
+        """
+        # Generate the structure for binary erosion and dilation, which is used to remove noise-like pixels.
+        # Groups of pixels with a radius smaller than ``regularizationWidth``
+        # will be excluded from regularization.
+        image = maskedImage.image.array
+        filterStructure = ndimage.iterate_structure(ndimage.generate_binary_structure(2, 1),
+                                                    regularizationWidth)
+        if highThreshold is not None:
```

Comment:
This can be just `if highThreshold:` as None is always False-y, and you are only checking that it is not None anyway, not that it is something specific.

Reply:
Actually, I had to revert this change. Since `highThreshold` is an array, `if highThreshold` throws an error.